src/openloco/windows/CompanyWindow.cpp

```diff
@@ -0,0 +1,1544 @@
+        make_widget({ 0, 14 }, { 26, 26 }, widget_type::wt_9, 0, string_ids::null, string_ids::tooltip_select_company)
+
+        // 0x004343FC
+        static void toggleRenameCompany(window* self)
+        {
+            self->disabled_widgets = 0;
+            if (self->number != companymgr::get_controlling_id())
+                self->disabled_widgets |= (1 << widx::tab_challenge);
+        }
+
+        // 0x00498E9B
+        static void enableRenameByCaption(window* self)
+        {
+            self->enabled_widgets |= (1 << caption);
+            if (!is_editor_mode())
+                self->enabled_widgets &= ~(1 << caption);
+        }
+
+        // Defined at the bottom of this file.
+        static void initEvents();
+        static void renameCompanyPrompt(window* self, widget_index widgetIndex);
+        static void switchTab(window* self, widget_index widgetIndex);
+        static void switchTabWidgets(window* self);
+        static void drawTabs(window* self, gfx::drawpixelinfo_t* dpi);
+        static void repositionTabs(window* self);
+    }
+
+    namespace status
+    {
+        static const gfx::ui_size_t windowSize = { 270, 182 };
+
+        enum widx
+        {
+            unk_11 = 11,
+            viewport,
+            centre_on_viewport,
+            face,
+            change_owner_name,
+        };
+
+        [[maybe_unused]] static widget_t widgets[] = {
+            commonWidgets(270, 182, string_ids::title_company),
+            make_widget({ 3, 160 }, { 242, 21 }, widget_type::wt_13, 1),
+            make_widget({ 3, 44 }, { 96, 120 }, widget_type::viewport, 1, -2),
+            make_widget({ 0, 0 }, { 24, 24 }, widget_type::wt_9, 1, string_ids::null, string_ids::move_main_view_to_show_this),
+            make_widget({ 178, 57 }, { 66, 66 }, widget_type::wt_9, 1, string_ids::null),
+            make_widget({ 154, 124 }, { 112, 22 }, widget_type::wt_9, 1, string_ids::null, string_ids::tooltip_change_owner_name),
+            widget_end(),
+        };
+
+        const uint64_t enabledWidgets = common::enabledWidgets | (1 << widx::centre_on_viewport) | (1 << widx::face) | (1 << widx::change_owner_name);
+
+        static window_event_list events;
+
+        // 0x00431EBB
+        static void prepare_draw(window* self)
+        {
+            printf("In prepare draw for company window %p with number = %d\n", self, self->number);
+
+            common::switchTabWidgets(self);
+
+            // Set company name.
+            auto company = companymgr::get(self->number);
+            *_common_format_args = company->name;
+
+            self->disabled_widgets &= ~((1 << widx::centre_on_viewport) | (1 << widx::face));
+
+            // No centering on a viewport that doesn't exist.
+            // if (self->viewports[0] == nullptr)
+            //     self->disabled_widgets |= (1 << widx::centre_on_viewport);
```

Comment:
With this line uncommented, the game crashes when drawing the widgets for this window.

Reply:
The issue is that `wt_9` doesn't use text content, it has an image (u32). string_ids::null is an `u16`, causing the game to try and render image 65535. So the widget definition is wrong.

Reply:
Hmm, interesting. The definition is set to `image_ids::null` though, which is an `uint32_t` set to `0xFFFFFFFF`. In effect the same thing, I guess. The widget definitions were adapted from the binary though, so I'm not sure how to change them. Perhaps the rendering code needs a guard instead?

(Curiously, the same definition is used in the town window without problems — but that doesn't appear to be disabled at any point.)

Reply:
No, the definition is set to `string_ids::null`, which evaluates to `0x0000FFFF`

Reply:
I understand the difference, haha. You were right, though, it was set to `string_ids::null`. I must have been looking at the wrong definition last night. Fixed now.